### PR TITLE
Stream-writer for a rope

### PR DESCRIPTION
**Pull request by data-man**

### CODE REVIEW

**Comment by data-man** [comment]
Along the way, I found this bug:
```nim
var s = newStringStream("The quick brown fox jumped over the lazy dog. The lazy dog ran.")
write(s, "123")
echo s.peekLine

File: lib/pure/ropes.nim [hunk 1]
```diff
@@ -130,7 +131,7 @@ proc insertInCache(s: string, tree: Rope): Rope =
       result.left = t
       t.right = nil
 
-proc rope*(s: string): Rope {.rtl, extern: "nro$1Str".} =
+proc rope*(s: string = ""): Rope {.rtl, extern: "nro$1Str".} =
```
Comment (Araq) [inline lib/pure/ropes.nim]: ``nil`` is the empty rope.

**Comment by data-man** [comment]
@andreaferretti 
Try to run this code, but first guess what should be output?

File: lib/pure/ropes.nim (hunk 1 above)
Comment (data-man) [reply lib/pure/ropes.nim]: Done.

**Comment by Yardanico** [comment]
@data-man why are you asking @andreaferretti ?

File: lib/pure/ropes.nim [hunk 2]
```diff
@@ -130,7 +131,7 @@ proc insertInCache(s: string, tree: Rope): Rope =
       result.left = t
       t.right = nil
 
-proc rope*(s: string): Rope {.rtl, extern: "nro$1Str".} =
+proc rope*(s: string = nil): Rope {.rtl, extern: "nro$1Str".} =
```
Comment (Araq) [inline lib/pure/ropes.nim]: No, an empty rope is ``nil`` not the string, just remove the default value here.

**Comment by data-man** [comment]
@Yardanico 
Because @andreaferretti reacted with confused emoji ;)

File: lib/pure/ropes.nim (hunk 2 above)
Comment (data-man) [reply lib/pure/ropes.nim]: Oh, I added it to be able to create a emty rope.
Because newRope is not exported.

**Pull request merged**